### DISCUSSION TITLE
Intermittent 500 errors on a minimal API route

### DISCUSSION THREAD

**Discussion by jnorris441** [vercel/next.js#41105] [Help]
Hi, I am posting some JSON to this endpoint. About a quarter of the time I get a 500 error. Seems like an issue with Vercel and not this code.

```
import type { NextApiRequest, NextApiResponse } from 'next'

export default async function handler(
    req: NextApiRequest,
    res: NextApiResponse<any>
) {

    res.json({ body: req.body, cookies: req.cookies })

}
```

**Comment by icyJoseph** [comment]
>  I am posting some JSON to this endpoint. 

What's the size of this JSON payload? Is it correctly shaped JSON? What's the size of cookies?

> About a quarter of the time I get a 500 error.

How often are you posting the JSON payloads? 

There's usage limits on Vercel's API routes, which you can read about there: https://vercel.com/docs/concepts/limits/overview, and crossing those often results in 5xx's but with the amount of info I got this is as far as I can go.

**Reply by jnorris441** [reply]
@icyJoseph thanks for the reply. 

The body is very short just to test that it sends something back. I was testing without sending a cookie.

I just tested in Postman clicking once every couple seconds. I can't imagine that triggered a rate limit.

**Reply by jnorris441** [reply]
an example body:

```
{"accessToken":"eyJhbGciOiJIUzI1...rest of token"}
```

**Reply by icyJoseph** [reply]
Is there any more data on those 5xx's? I cannot think of any 'technical' reason why the API would just reject. 

Getting a bit creative, perhaps different lambdas are being created, and then you hit the soft-limit of 12 different lambdas (hobby-plan), though, AFAIK, Vercel tries to group as many functions as possible into one lambda. https://vercel.com/docs/concepts/limits/overview#serverless-functions-created-per-deployment

Are there any logs in the Vercel deployment that we could go through?

**Reply by jnorris441** [reply]
@icyJoseph https://vercel.com/jnorris441/catalog-site/CupsLdgiTZYSKEg1S92dopnJJWhS/functions?name=api%2Ftest

I was trying to fix 500 errors from /api/servers and was outputting a debug string. Then I made /api/test with the minimal code above.

When testing /api/test it was outputting the debug string from /api/servers in the 500 error for some reason. Seemed mixed up.

**Reply by icyJoseph** [reply]
Ah I can't see those logs because they are private :) no worries though. 

I am not sure what the issue could be though. Perhaps it is best to take it with the Vercel support or something like that.

**Reply by jnorris441** [reply]
Thanks. Will take 3 or 5 days to hear back though. I just migrated the API away from Vercel since it doesn't seem reliable for this project. Will be interested to see what they say because I would want to use it in the future.

**Reply by icyJoseph** [reply]
Yeah, I haven't really experienced anything like this. I definitely would like to know what's the resolution.
